packages/x-data-grid/src/material/variables.ts

```diff
@@ -36,27 +37,29 @@ function transformTheme(t: Theme): GridCSSVariablesInterface {
 
   const fontBody = (t.vars as any)?.font?.body2 ?? formatFont(t.typography.body2);
   const fontSmall = (t.vars as any)?.font?.caption ?? formatFont(t.typography.caption);
+  const fontLarge = (t.vars as any)?.font?.body1 ?? formatFont(t.typography.body1);
 
   const k = vars.keys;
 
   return {
     [k.spacingUnit]: t.vars ? ((t.vars as any).spacing ?? t.spacing(1)) : t.spacing(1),
 
     [k.colors.border.base]: borderColor,
+    [k.colors.border.muted]: mutedBorderColor,
     [k.colors.background.base]: backgroundBase,
     [k.colors.background.overlay]: backgroundOverlay,
     [k.colors.background.backdrop]: backgroundBackdrop,
-    [k.colors.foreground.base]: (t.vars || t).palette.text.primary,
-    [k.colors.foreground.muted]: (t.vars || t).palette.text.secondary,
-    [k.colors.foreground.accent]: (t.vars || t).palette.primary.dark,
-    [k.colors.foreground.disabled]: (t.vars || t).palette.text.disabled,
-
-    [k.colors.interactive.hover]: removeOpacity((t.vars || t).palette.action.hover),
-    [k.colors.interactive.hoverOpacity]: (t.vars || t).palette.action.hoverOpacity,
-    [k.colors.interactive.focus]: removeOpacity((t.vars || t).palette.primary.main),
-    [k.colors.interactive.focusOpacity]: (t.vars || t).palette.action.focusOpacity,
-    [k.colors.interactive.disabled]: removeOpacity((t.vars || t).palette.action.disabled),
-    [k.colors.interactive.disabledOpacity]: (t.vars || t).palette.action.disabledOpacity,
+    [k.colors.foreground.base]: t.palette.text.primary,
+    [k.colors.foreground.muted]: t.palette.text.secondary,
+    [k.colors.foreground.accent]: t.palette.primary.dark,
+    [k.colors.foreground.disabled]: t.palette.text.disabled,
+
+    [k.colors.interactive.hover]: t.palette.action.hover,
+    [k.colors.interactive.hoverOpacity]: t.palette.action.hoverOpacity,
+    [k.colors.interactive.focus]: t.palette.primary.main,
+    [k.colors.interactive.focusOpacity]: t.palette.action.focusOpacity,
+    [k.colors.interactive.disabled]: t.palette.action.disabled,
+    [k.colors.interactive.disabledOpacity]: t.palette.action.disabledOpacity,
```

Comment:
Changes done in #16962 seem to be reverted here, is this intentional?

Reply:
Must have been a bad merge, thanks for picking up on it. https://github.com/mui/mui-x/pull/9877/commits/46a20080d069d543f18193ee1a52ad7181a626df